File: profiles/lib/destination.lua
local Destination = {}

function Destination.get_directional_tag(way, is_forward, tag)
  local v
  if is_forward then
    v = way:get_value_by_key(tag .. ':forward') or way:get_value_by_key(tag)
  else
    v = way:get_value_by_key(tag .. ':backward') or way:get_value_by_key(tag)
  end
  if v then
    return v.gsub(v, ';', ', ')
  end
end

-- Assemble destination as: "A59: Düsseldorf, Köln"
--          destination:ref  ^    ^  destination

function Destination.get_destination(way, is_forward)
  ref  = Destination.get_directional_tag(way, is_forward, 'destination:ref')
  dest = Destination.get_directional_tag(way, is_forward, 'destination')
  street = Destination.get_directional_tag(way, is_forward, 'destination:street')
  if ref and dest then
    return ref .. ': ' .. dest
  else
    return ref or dest or street or ''
  end
end

return Destination

File: profiles/lib/destination.js
module.exports = function(way) {
  const destination = way.tag("destination");
  const destination_ref = way.tag("destination:ref");

  // Assemble destination as: "A59: Düsseldorf, Köln"
  //          destination:ref  ^    ^  destination

  let rv = "";

  if (destination_ref) {
    rv = destination_ref.replace(/;/g, ', ');
  }

  if (destination) {
      if (rv) {
          rv += ": "
      }

      rv += destination.replace(/;/g, ', ');
  }

  return rv;
};
